as/plugins/org.jboss.ide.eclipse.as.wtp.core/src/org/jboss/ide/eclipse/as/core/server/UnitedServerListenerManager.java

```diff
@@ -48,6 +48,14 @@ public static synchronized UnitedServerListenerManager getDefault() {
 	protected ArrayList<UnitedServerListener> list;
 	private UnitedServerListenerManager() {
 		list = new ArrayList<UnitedServerListener>();
+		new Thread() {

```

Comment:
By moving the code that interacts with wtp into a new thread which is not synchronized, we ensure that in this class, there is no code that needs both the lock of this class, AND the lock of any specific server.   The code in this PR only needs 1 lock at a time.  The constructor as previously written could easily require multiple locks and then deadlock based on the behavior of other threads.